lib/platform/bitbucket-server/index.js

```diff
@@ -611,18 +611,38 @@ async function updatePr(prNo, title, description) {
 
 async function mergePr(prNo) {
   logger.debug(`mergePr(${prNo})`);
-  // TODO: Needs implementation
   // Used for "automerge" feature
-  const { version } = (await api.get(
-    `./rest/api/1.0/projects/${config.projectKey}/repos/${
-      config.repositorySlug
-    }/pull-requests/${prNo}`
-  )).body;
+
+  const pr = await getPr(prNo);
+
+  // istanbul ignore if
+  if (!pr) {
+    logger.warn(`Failed to merge PR ${prNo} -- missing`);
+    return false;
+  }
+
+  // istanbul ignore if
+  if (pr.state != 'open') {
```

Comment:
This state probably shouldn't happen, but it's OK to add this check. At an application level (e.g. lib/workers code) it shouldn't attempt to merge a PR that the platform hasn't reported as 'open'. We do cache PRs however, and it's possible that a PR gets closed by a user or another process after we originally fetched it, but in such cases we should probably try/catch and then throw a `repository-changed` Error. Doing so aborts the process because our state is out of date and may even be completely wrong.

Reply:
Ok